scheduler/src/cook/rest/api.clj

```diff
@@ -920,13 +918,19 @@
                 (assoc :partitions (into #{} (walk/stringify-keys partitions))))))
        (into #{})))
 
-(defn validate-job-gpu-model
-  "Validates that GPU model entered is supported on the pool"
-  [pool-name {:keys [env]}]
-  (let [requested-gpu-model (get env "COOK_GPU_MODEL")]
+(defn validate-gpu-job
+  "Validates that a job requesting GPUs is supported on the pool"
+  [gpu-enabled? pool-name {:keys [gpus env]}]
+  (let [requested-gpu-model (get env "COOK_GPU_MODEL")
+        gpus' (or gpus 0)]
+    (when (and (pos? gpus') (not gpu-enabled?))
+      (throw (ex-info (str "GPU support is not enabled") {})))
```

Comment:
Should we add a case to the integration tests for this?

Reply:
this case is already incorporated in integration tests